pkg/validator/validators.go

```diff
@@ -111,10 +111,27 @@ func ingressBackendValidator(req *admissionv1.AdmissionRequest) (*admissionv1.Ad
 	for _, source := range ingressBackend.Spec.Sources {
 		switch source.Kind {
 		// Add validation for source kinds here
+		case policyv1alpha1.KindService:
+			if source.Name == "" {
+				return nil, errors.Errorf("'source.name' not specified for source kind %s", policyv1alpha1.KindService)
+			}
+			if source.Namespace == "" {
+				return nil, errors.Errorf("'source.namespace' not specified for source kind %s", policyv1alpha1.KindService)
+			}
+
+		case policyv1alpha1.KindAuthenticatedPrincipal:
+			if source.Name == "" {
+				return nil, errors.Errorf("'source.name' not specified for source kind %s", policyv1alpha1.KindAuthenticatedPrincipal)
+			}
+
 		case policyv1alpha1.KindIPRange:
 			if _, _, err := net.ParseCIDR(source.Name); err != nil {
-				return nil, errors.Errorf("Invalid 'source.Name' value specified for IPRange. Expected CIDR notation 'a.b.c.d/x', got '%s'", source.Name)
+				return nil, errors.Errorf("Invalid 'source.name' value specified for IPRange. Expected CIDR notation 'a.b.c.d/x', got '%s'", source.Name)
 			}
+
+		default:
+			return nil, errors.Errorf("Invalid 'source.kind' value specified. Must be one of: %s, %s, %s",
+				policyv1alpha1.KindService, policyv1alpha1.KindAuthenticatedPrincipal, policyv1alpha1.KindIPRange)
```

Comment:
One alternative idea that comes to mind is to borrow a principle from FP and create separate validator functions per kind. Then depending on the kind type we apply the given function. This might make testing and adding new validators in the future easier.